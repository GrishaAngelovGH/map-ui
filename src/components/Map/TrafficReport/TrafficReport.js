import Modal from 'react-bootstrap/Modal'

const TrafficReport = ({ show, onHide }) => (
  <Modal show={show} onHide={onHide}>
    <Modal.Header closeButton>
      <Modal.Title>Modal Title</Modal.Title>
    </Modal.Header>
    <Modal.Body>Modal Body</Modal.Body>
  </Modal>
)

export default TrafficReport