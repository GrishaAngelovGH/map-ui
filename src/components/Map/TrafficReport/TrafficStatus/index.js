import TrafficStatus from './TrafficStatus'

export default TrafficStatus