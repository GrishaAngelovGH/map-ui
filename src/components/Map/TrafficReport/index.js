import TrafficReport from './TrafficReport'

export default TrafficReport