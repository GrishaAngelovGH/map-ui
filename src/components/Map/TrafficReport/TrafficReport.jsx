import Modal from 'react-bootstrap/Modal'
import TrafficStatus from './TrafficStatus'

const TrafficReport = ({ show, onHide }) => (
  <Modal show={show} onHide={onHide}>
    <Modal.Header closeButton>
      <Modal.Title>
        <i className='bi bi-geo-alt-fill text-primary border border-3 border-primary rounded'></i>
        <span className='mx-2'>91 Knightsbridge, London</span>
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <p className='m-0 text-center'>91 Knightsbridge, London SW1X 8ET</p>
      <p className='m-0 text-secondary text-center'>Fastest route now due to traffic conditions</p>

      <TrafficStatus status='low' />
      <TrafficStatus status='high' />
      <TrafficStatus status='normal' />
    </Modal.Body>
  </Modal>
)

export default TrafficReport