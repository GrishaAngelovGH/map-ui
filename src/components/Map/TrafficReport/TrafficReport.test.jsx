import { render } from '@testing-library/react'

import TrafficReport from './TrafficReport'

test('should render TrafficReport component', () => {
  const view = render(<TrafficReport show />)

  expect(view).toMatchSnapshot()
})