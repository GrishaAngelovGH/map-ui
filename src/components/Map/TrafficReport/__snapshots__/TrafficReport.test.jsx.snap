// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render TrafficReport component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              <i
                class="bi bi-geo-alt-fill text-primary border border-3 border-primary rounded"
              />
              <span
                class="mx-2"
              >
                91 Knightsbridge, London
              </span>
            </div>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            <p
              class="m-0 text-center"
            >
              91 Knightsbridge, London SW1X 8ET
            </p>
            <p
              class="m-0 text-secondary text-center"
            >
              Fastest route now due to traffic conditions
            </p>
            <div
              class="row mt-3 justify-content-center"
            >
              <div
                class="col-md-10 border rounded shadow"
              >
                <div
                  class="row small"
                >
                  <div
                    class="col-md-4 p-3"
                  >
                    <div
                      class="alert alert-warning m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
                    >
                      <p
                        class="m-0"
                      >
                        9 min
                      </p>
                      <p
                        class="m-0"
                      >
                        (1.9 km)
                      </p>
                    </div>
                  </div>
                  <div
                    class="col-md-8 p-1"
                  >
                    <p
                      class="m-0 text-success text-center fw-bold"
                    >
                      Via A4 and A308
                    </p>
                    <p
                      class="alert alert-warning text-center fw-bold m-0 p-1"
                    >
                      Low Traffic
                    </p>
                    <p
                      class="m-0 text-secondary d-flex align-items-center"
                    >
                      <i
                        class="bi bi-sign-turn-slight-right-fill text-primary me-2"
                      />
                      Next Direction
                    </p>
                    <p
                      class="m-0"
                    >
                      Head west on Knightsbridge
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row mt-3 justify-content-center"
            >
              <div
                class="col-md-10 border rounded shadow"
              >
                <div
                  class="row small"
                >
                  <div
                    class="col-md-4 p-3"
                  >
                    <div
                      class="alert alert-danger m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
                    >
                      <p
                        class="m-0"
                      >
                        9 min
                      </p>
                      <p
                        class="m-0"
                      >
                        (1.9 km)
                      </p>
                    </div>
                  </div>
                  <div
                    class="col-md-8 p-1"
                  >
                    <p
                      class="m-0 text-success text-center fw-bold"
                    >
                      Via A4 and A308
                    </p>
                    <p
                      class="alert alert-danger text-center fw-bold m-0 p-1"
                    >
                      High Traffic
                    </p>
                    <p
                      class="m-0 text-secondary d-flex align-items-center"
                    >
                      <i
                        class="bi bi-sign-turn-slight-right-fill text-primary me-2"
                      />
                      Next Direction
                    </p>
                    <p
                      class="m-0"
                    >
                      Head west on Knightsbridge
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row mt-3 justify-content-center"
            >
              <div
                class="col-md-10 border rounded shadow"
              >
                <div
                  class="row small"
                >
                  <div
                    class="col-md-4 p-3"
                  >
                    <div
                      class="alert alert-success m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
                    >
                      <p
                        class="m-0"
                      >
                        9 min
                      </p>
                      <p
                        class="m-0"
                      >
                        (1.9 km)
                      </p>
                    </div>
                  </div>
                  <div
                    class="col-md-8 p-1"
                  >
                    <p
                      class="m-0 text-success text-center fw-bold"
                    >
                      Via A4 and A308
                    </p>
                    <p
                      class="alert alert-success text-center fw-bold m-0 p-1"
                    >
                      Normal Traffic
                    </p>
                    <p
                      class="m-0 text-secondary d-flex align-items-center"
                    >
                      <i
                        class="bi bi-sign-turn-slight-right-fill text-primary me-2"
                      />
                      Next Direction
                    </p>
                    <p
                      class="m-0"
                    >
                      Head west on Knightsbridge
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
